do_everything.sh

```diff
@@ -378,31 +382,20 @@ echo
 
 if [[ $debugging -eq 1 ]];then
     echo "Build type = DEBUG"
-    run_cmd build_cpp -p $prefix -b Debug -v $verbose
+    run_cmd build_cpp -p $prefix/catkin_ws -r $TARGET_DIR -b Debug -v $verbose
 else
     echo "Build type = RELEASE"
-    run_cmd build_cpp -p $prefix -b Release -v $verbose
+    run_cmd build_cpp -p $prefix/catkin_ws -r $TARGET_DIR -b Release -v $verbose
```

Comment:
In line with my proposal above, here I'd send `prefix` and `prefix/catkin_ws`, and let the script figure `TARGET_DIR` as `prefix/target` as it was before.